File: yntra-android/app/src/main/java/com/yntra/app/views/AuthView.kt
package com.yntra.app.views

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yntra.app.viewmodels.AuthViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AuthView(viewModel: AuthViewModel) {
    val isLoggedIn by viewModel.isLoggedIn.collectAsState()
    val authRegion by viewModel.authRegion.collectAsState()
    val bankIdFlowState by viewModel.bankIdFlowState.collectAsState()
    val qrData by viewModel.qrData.collectAsState()
    val errorMessage by viewModel.errorMessage.collectAsState()

    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var showPasswordLogin by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF0B0F19))
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        // App Brand Header
        Spacer(modifier = Modifier.height(40.dp))
        Icon(
            imageVector = Icons.Default.Lock,
            contentDescription = "Lock",
            tint = Color(0xFF8B5CF6),
            modifier = Modifier.size(64.dp)
        )
        Text(
            text = "Yntra Secure Gate",
            color = Color.White,
            fontSize = 28.sp,
            fontWeight = FontWeight.Black
        )
        Text(
            text = "Please authenticate to access your local workspace.",
            color = Color(0xFF94A3B8),
            fontSize = 14.sp
        )

        // Error message card
        errorMessage?.let {
            Card(
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.errorContainer),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth()
            ) {
                Row(
                    modifier = Modifier.padding(16.dp),
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Icon(imageVector = Icons.Default.Warning, contentDescription = "Error", tint = MaterialTheme.colorScheme.error)
                    Text(text = it, color = MaterialTheme.colorScheme.onErrorContainer, fontSize = 14.sp)
                }
            }
        }

        if (bankIdFlowState == "qr_scan" || bankIdFlowState == "connecting") {
            // BankID flow popup/container
            Card(
                colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
                shape = RoundedCornerShape(24.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 12.dp)
            ) {
                Column(
                    modifier = Modifier.padding(24.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    Text(
                        text = if (bankIdFlowState == "connecting") "Connecting to Identity Provider..." else "Scan QR Code",
                        color = Color.White,
                        fontWeight = FontWeight.Bold,
                        fontSize = 18.sp
                    )
                    
                    if (bankIdFlowState == "qr_scan" && qrData != null) {
                        // QR Code placeholder using custom drawing or standard box
                        Box(
                            modifier = Modifier
                                .size(200.dp)
                                .background(Color.White, RoundedCornerShape(16.dp))
                                .border(4.dp, Color(0xFF8B5CF6), RoundedCornerShape(16.dp))
                                .padding(16.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = "QR: ${qrData!!.take(12)}...",
                                color = Color.Black,
                                fontSize = 12.sp,
                                fontWeight = FontWeight.Bold
                            )
                        }
                    } else {
                        CircularProgressIndicator(color = Color(0xFF4F46E5))
                    }

                    Button(
                        onClick = { viewModel.logout() },
                        colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFEF4444)),
                        shape = RoundedCornerShape(16.dp)
                    ) {
                        Text("Cancel")
                    }
                }
            }
        } else {
            // Regular auth selections
            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                // Region buttons
                Text(
                    text = "Select authentication region",
                    color = Color(0xFF94A3B8),
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Bold
                )
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    listOf("sv" to "SE", "da" to "DK", "no" to "NO", "en" to "US").forEach { (code, label) ->
                        val isSelected = authRegion == code
                        Box(
                            modifier = Modifier
                                .weight(1f)
                                .background(
                                    if (isSelected) Color(0xFF4F46E5) else Color(0xFF1E293B),
                                    RoundedCornerShape(12.dp)
                                )
                                .clickable { viewModel.setRegion(code) }
                                .padding(vertical = 12.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(text = label, color = Color.White, fontWeight = FontWeight.Bold)
                        }
                    }
                }

                Spacer(modifier = Modifier.height(8.dp))

                // eID buttons
                Button(
                    onClick = {
                        val provider = when (authRegion) {
                            "sv" -> "se_bankid"
                            "da" -> "dk_mitid"
                            "no" -> "no_bankid"
                            else -> "us_global"
                        }
                        viewModel.initiateBankIdLogin(provider)
                    },
                    modifier = Modifier.fillMaxWidth(),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF4F46E5)),
                    shape = RoundedCornerShape(16.dp),
                    contentPadding = PaddingValues(16.dp)
                ) {
                    Text(
                        text = when (authRegion) {
                            "sv" -> "Login with Mobilt BankID"
                            "da" -> "Login with MitID"
                            "no" -> "Login with BankID"
                            else -> "Login with National eID"
                        },
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Bold
                    )
                }

                Button(
                    onClick = {
                        viewModel.initiateBankIdLogin("card_or_badge")
                    },
                    modifier = Modifier.fillMaxWidth(),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF1E293B)),
                    shape = RoundedCornerShape(16.dp),
                    contentPadding = PaddingValues(16.dp)
                ) {
                    Text(text = "Authenticate via SITHS / NFC Badge", fontSize = 14.sp)
                }

                Spacer(modifier = Modifier.height(16.dp))

                // Credentials fallback
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { showPasswordLogin = !showPasswordLogin }
                        .padding(vertical = 8.dp),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = if (showPasswordLogin) "Hide Credentials Fallback" else "Use Credentials Fallback",
                        color = Color(0xFF8B5CF6),
                        fontSize = 14.sp,
                        fontWeight = FontWeight.SemiBold
                    )
                }

                AnimatedVisibility(visible = showPasswordLogin) {
                    Column(
                        modifier = Modifier.fillMaxWidth(),
                        verticalArrangement = Arrangement.spacedBy(10.dp)
                    ) {
                        OutlinedTextField(
                            value = email,
                            onValueChange = { email = it },
                            label = { Text("Email / Username") },
                            modifier = Modifier.fillMaxWidth(),
                            colors = OutlinedTextFieldDefaults.colors(
                                focusedBorderColor = Color(0xFF8B5CF6),
                                unfocusedBorderColor = Color(0xFF1E293B),
                                focusedContainerColor = Color(0xFF1E293B),
                                unfocusedContainerColor = Color(0xFF1E293B),
                                focusedTextColor = Color.White,
                                unfocusedTextColor = Color.White
                            )
                        )
                        OutlinedTextField(
                            value = password,
                            onValueChange = { password = it },
                            label = { Text("PIN / Password") },
                            modifier = Modifier.fillMaxWidth(),
                            colors = OutlinedTextFieldDefaults.colors(
                                focusedBorderColor = Color(0xFF8B5CF6),
                                unfocusedBorderColor = Color(0xFF1E293B),
                                focusedContainerColor = Color(0xFF1E293B),
                                unfocusedContainerColor = Color(0xFF1E293B),
                                focusedTextColor = Color.White,
                                unfocusedTextColor = Color.White
                            )
                        )
                        Button(
                            onClick = { viewModel.passwordLogin(email, password) },
                            modifier = Modifier.fillMaxWidth(),
                            colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF8B5CF6)),
                            shape = RoundedCornerShape(16.dp)
                        ) {
                            Text("Submit Credentials", fontWeight = FontWeight.Bold)
                        }
                    }
                }
            }
        }
    }
}

File: yntra-ios/YntraIOS/Views/AuthView.swift
import SwiftUI
import yntra_core
import CoreImage.CIFilterBuiltins

struct AuthView: View {
    @ObservedObject var viewModel: AuthViewModel
    
    @State private var email: String = ""
    @State private var pin: String = ""
    @State private var showCredentialsFallback: Bool = false
    
    // Brand Colors
    private let darkBackground = Color(red: 0.04, green: 0.06, blue: 0.1)
    private let cardBackground = Color(red: 0.12, green: 0.16, blue: 0.23)
    private let primaryGradient = LinearGradient(
        colors: [Color(red: 0.31, green: 0.27, blue: 0.9), Color(red: 0.54, green: 0.36, blue: 0.96)],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    var body: some View {
        ZStack {
            darkBackground.ignoresSafeArea()
            
            // Glowing Ambient Bubbles
            VStack {
                Circle()
                    .fill(Color(red: 0.31, green: 0.27, blue: 0.9).opacity(0.15))
                    .frame(width: 300, height: 300)
                    .blur(radius: 50)
                    .offset(x: -80, y: -50)
                Spacer()
            }
            .ignoresSafeArea()

            VStack(spacing: 24) {
                Spacer()
                
                // Brand Logo/Header
                Image(systemName: "lock.shield.fill")
                    .font(.system(size: 64))
                    .foregroundStyle(primaryGradient)
                    .shadow(color: Color(red: 0.31, green: 0.27, blue: 0.9).opacity(0.4), radius: 10, y: 5)
                
                VStack(spacing: 8) {
                    Text("Yntra Secure Gate")
                        .font(.system(size: 28, weight: .black, design: .rounded))
                        .foregroundColor(.white)
                    
                    Text("Please authenticate to access your local workspace.")
                        .font(.system(size: 14, weight: .medium, design: .rounded))
                        .foregroundColor(.white.opacity(0.6))
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 32)
                }
                
                // Error banner
                if let errorMessage = viewModel.errorMessage {
                    HStack {
                        Image(systemName: "exclamationmark.triangle.fill")
                            .foregroundColor(.red)
                        Text(errorMessage)
                            .font(.system(size: 13, weight: .semibold, design: .rounded))
                            .foregroundColor(.white)
                    }
                    .padding()
                    .background(Color.red.opacity(0.2))
                    .cornerRadius(14)
                    .overlay(
                        RoundedRectangle(cornerRadius: 14)
                            .stroke(Color.red.opacity(0.3), lineWidth: 1)
                    )
                    .padding(.horizontal)
                }
                
                if viewModel.bankIdFlowState == "qr_scan" || viewModel.bankIdFlowState == "connecting" {
                    // BankID Scan Popup view
                    VStack(spacing: 20) {
                        Text(viewModel.bankIdFlowState == "connecting" ? "Connecting to Provider..." : "Scan QR Code")
                            .font(.system(size: 18, weight: .bold, design: .rounded))
                            .foregroundColor(.white)
                        
                        if viewModel.bankIdFlowState == "qr_scan", let qrData = viewModel.qrData {
                            // QR code box
                            ZStack {
                                RoundedRectangle(cornerRadius: 20)
                                    .fill(Color.white)
                                    .frame(width: 200, height: 200)
                                    .shadow(radius: 10)
                                
                                Image(uiImage: generateQRCode(from: qrData))
                                    .interpolation(.none)
                                    .resizable()
                                    .scaledToFit()
                                    .frame(width: 180, height: 180)
                            }
                        } else {
                            ProgressView()
                                .tint(.white)
                        }
                        
                        Button(action: {
                            viewModel.logout()
                        }) {
                            Text("Cancel")
                                .bold()
                                .padding(.horizontal, 24)
                                .padding(.vertical, 12)
                                .background(Color.red)
                                .foregroundColor(.white)
                                .cornerRadius(12)
                        }
                    }
                    .padding(30)
                    .background(cardBackground)
                    .cornerRadius(24)
                    .overlay(
                        RoundedRectangle(cornerRadius: 24)
                            .stroke(Color.white.opacity(0.1), lineWidth: 1)
                    )
                    .padding(.horizontal)
                } else {
                    // Selection view
                    VStack(spacing: 16) {
                        // Region Selector
                        HStack(spacing: 8) {
                            ForEach([("sv", "SE"), ("da", "DK"), ("no", "NO"), ("en", "US")], id: \.0) { code, label in
                                Button(action: { viewModel.setRegion(code) }) {
                                    Text(label)
                                        .font(.system(size: 14, weight: .bold, design: .rounded))
                                        .foregroundColor(.white)
                                        .frame(maxWidth: .infinity)
                                        .padding(.vertical, 12)
                                        .background(viewModel.authRegion == code ? Color.blue : cardBackground)
                                        .cornerRadius(12)
                                }
                            }
                        }
                        .padding(.horizontal)
                        
                        // Action buttons
                        Button(action: {
                            let provider = viewModel.authRegion == "sv" ? "se_bankid" : (viewModel.authRegion == "da" ? "dk_mitid" : "no_bankid")
                            viewModel.initiateBankIdLogin(provider: provider)
                        }) {
                            Text(viewModel.authRegion == "sv" ? "Login with Mobilt BankID" : (viewModel.authRegion == "da" ? "Login with MitID" : "Login with BankID"))
                                .font(.system(size: 16, weight: .bold, design: .rounded))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 16)
                                .background(primaryGradient)
                                .cornerRadius(16)
                                .shadow(color: Color(red: 0.31, green: 0.27, blue: 0.9).opacity(0.4), radius: 8, y: 4)
                        }
                        .padding(.horizontal)
                        
                        Button(action: {
                            viewModel.initiateBankIdLogin(provider: "card_or_badge")
                        }) {
                            Text("Authenticate via SITHS / NFC Badge")
                                .font(.system(size: 14, weight: .semibold, design: .rounded))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 14)
                                .background(cardBackground)
                                .cornerRadius(16)
                        }
                        .padding(.horizontal)
                        
                        // Credentials fallback
                        Button(action: {
                            withAnimation {
                                showCredentialsFallback.toggle()
                            }
                        }) {
                            Text(showCredentialsFallback ? "Hide Credentials Fallback" : "Use Credentials Fallback")
                                .font(.system(size: 13, weight: .bold, design: .rounded))
                                .foregroundColor(.purple)
                        }
                        .padding(.top, 8)
                        
                        if showCredentialsFallback {
                            VStack(spacing: 12) {
                                TextField("Email / Username", text: $email)
                                    .textFieldStyle(PlainTextFieldStyle())
                                    .padding()
                                    .background(cardBackground)
                                    .cornerRadius(14)
                                    .foregroundColor(.white)
                                
                                SecureField("PIN / Password", text: $pin)
                                    .textFieldStyle(PlainTextFieldStyle())
                                    .padding()
                                    .background(cardBackground)
                                    .cornerRadius(14)
                                    .foregroundColor(.white)
                                
                                Button(action: {
                                    viewModel.passwordLogin(email: email, pin: pin)
                                }) {
                                    Text("Submit Credentials")
                                        .bold()
                                        .foregroundColor(.white)
                                        .frame(maxWidth: .infinity)
                                        .padding()
                                        .background(Color.purple)
                                        .cornerRadius(14)
                                }
                            }
                            .padding(.horizontal)
                            .transition(.move(edge: .bottom).combined(with: .opacity))
                        }
                    }
                }
                
                Spacer()
            }
        }
    }

    private func generateQRCode(from string: String) -> UIImage {
        let context = CIContext()
        let filter = CIFilter.qrCodeGenerator()
        filter.message = Data(string.utf8)

        if let outputImage = filter.outputImage {
            let transform = CGAffineTransform(scaleX: 10, scaleY: 10)
            let scaledImage = outputImage.transformed(by: transform)
            if let cgImage = context.createCGImage(scaledImage, from: scaledImage.extent) {
                return UIImage(cgImage: cgImage)
            }
        }

        return UIImage(systemName: "xmark.circle") ?? UIImage()
    }
}
